x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -347,98 +268,55 @@ export class MapsAppView extends React.Component {
     const globalFilters = filterManager.getGlobalFilters();
     const allFilters = [...savedQueryFilters, ...globalFilters];
 
-    if (savedQuery.attributes.timefilter) {
-      if (savedQuery.attributes.timefilter.refreshInterval) {
-        this._onRefreshChange({
-          isPaused: savedQuery.attributes.timefilter.refreshInterval.pause,
-          refreshInterval: savedQuery.attributes.timefilter.refreshInterval.value,
-        });
-      }
-      this._onQueryChange({
-        filters: allFilters,
-        query: savedQuery.attributes.query,
-        time: savedQuery.attributes.timefilter,
-      });
-    } else {
-      this._onQueryChange({
-        filters: allFilters,
-        query: savedQuery.attributes.query,
+    const refreshInterval = _.get(savedQuery, 'attributes.timefilter.refreshInterval');
+    if (refreshInterval) {
+      this._onRefreshConfigChange({
+        isPaused: refreshInterval.pause,
+        interval: refreshInterval.value,
       });
     }
+    this._onQueryChange({
+      filters: allFilters,
+      query: savedQuery.attributes.query,
+      time: savedQuery.attributes.timefilter,
+    });
   }
 
-  _syncStoreAndGetFilters() {
-    const {
-      savedMap,
-      setGotoWithCenter,
-      setMapSettings,
-      setIsLayerTOCOpen,
-      setOpenTOCDetails,
-    } = this.props;
```

Comment:
Why remove the destructuring assignment?

Reply:
Just adds a lot of noise. It also makes it more work to figure out where variables are coming from since you can not see where they are destructured in the same context of where they are used.

Reply:
Hmm. I would argue the succinctness you gain in the subsequent logic lessens the overall noise. That said, it's a style thing, not a deal-breaker either way.